client/layout/utils.ts

```diff
@@ -14,3 +14,114 @@ export function getShouldShowAppBanner( site: any ): boolean {
 	}
 	return true;
 }
+
+let lastScrollPosition = 0; // Used for calculating scroll direction.
+let sidebarTop = 0; // Current sidebar top position.
+let pinnedSidebarTop = true;
+let pinnedSidebarBottom = false;
+let ticking = false; // Used for Scroll event throttling.
```

Comment:
Is the purpose of `ticking` to indicate active scrolling?

Reply:
`ticking:true` indicates that we are currently animating or that a scroll happens while we are in the same animation frame. So when `ticking:true` we should not run all the calculations again. It is only used for performance reasons.